x-pack/plugins/cases/public/components/header_page/editable_title.tsx

```diff
@@ -47,92 +24,75 @@ const EditableTitleComponent: React.FC<EditableTitleProps> = ({ onSubmit, isLoad
   const { releasePhase, permissions } = useCasesContext();
   const [editMode, setEditMode] = useState(false);
   const [errors, setErrors] = useState<string[]>([]);
-  const [newTitle, setNewTitle] = useState<string>(title);
 
-  const onCancel = useCallback(() => {
-    setEditMode(false);
-    setErrors([]);
-    setNewTitle(title);
-  }, [title]);
+  const onClickSubmit = useCallback(
+    (newTitleValue: string): boolean => {
+      if (!newTitleValue.trim().length) {
+        setErrors([i18n.TITLE_REQUIRED]);
+        return false;
+      }
 
-  const onClickEditIcon = useCallback(() => setEditMode(true), []);
-  const onClickSubmit = useCallback((): void => {
-    if (!newTitle.trim().length) {
-      setErrors([i18n.TITLE_REQUIRED]);
-      return;
-    }
+      if (newTitleValue.trim().length > MAX_TITLE_LENGTH) {
+        setErrors([i18n.MAX_LENGTH_ERROR('title', MAX_TITLE_LENGTH)]);
+        return false;
+      }
 
-    if (newTitle.trim().length > MAX_TITLE_LENGTH) {
-      setErrors([i18n.MAX_LENGTH_ERROR('title', MAX_TITLE_LENGTH)]);
-      return;
-    }
+      if (newTitleValue !== title) {
+        onSubmit(newTitleValue);
+      }
+      setEditMode(false);
+      setErrors([]);
 
-    if (newTitle !== title) {
-      onSubmit(newTitle);
-    }
-    setEditMode(false);
-    setErrors([]);
-  }, [newTitle, onSubmit, title]);
+      return true;
+    },
+    [onSubmit, title]
+  );
 
-  const handleOnChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
-    setNewTitle(e.target.value);
+  const onCancel = () => {
     setErrors([]);
-  }, []);
+    setEditMode(false);
+  };
 
   const hasErrors = errors.length > 0;
 
-  return editMode ? (
-    <EuiFormRow isInvalid={hasErrors} error={errors} fullWidth>
-      <EuiFlexGroup
-        alignItems="center"
-        responsive={true}
-        gutterSize="m"
-        justifyContent="spaceBetween"
-      >
-        <EuiFlexItem grow={true}>
-          <EuiFieldText
-            fullWidth={true}
-            onChange={handleOnChange}
-            value={`${newTitle}`}
-            data-test-subj="editable-title-input-field"
-          />
-        </EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <EuiButton
-            color="success"
-            data-test-subj="editable-title-submit-btn"
-            fill
-            iconType="save"
-            onClick={onClickSubmit}
+  return (
+    <>
```

Comment:
nit: I think this fragment is unnecessary because we are only returning `EuiFlexGroup` 😅